executor/runtime/docker/docker.go

```diff
@@ -1997,8 +1997,15 @@ func (r *DockerRuntime) Kill(c *runtimeTypes.Container) error { // nolint: gocyc
 	}
 
 stopped:
+	l := log.WithField("taskId", c.TaskID)
 	if c.SetupCommand != nil && c.SetupCommand.Process != nil {
 		_ = c.SetupCommand.Process.Signal(unix.SIGTERM) // nolint: gosec
+		time.AfterFunc(30*time.Second, func() {
+			_ = c.SetupCommand.Process.Kill() // nolint: gosec
+		})
+		l.Info("Waiting for network teardown to finish")
+		_ = c.SetupCommand.Wait() // nolint: gosec
```

Comment:
Do we need to call `Stop()` on the timer returned by `time.AfterFunc()` here, or is it fine to let that function run when SIGTERM caused the process to exit?